src/main/java/org/snomed/snowstorm/core/data/domain/Relationship.java

```diff
@@ -187,30 +187,28 @@ public boolean isGrouped () {
 	}
 
 	public ConcreteValue getConcreteValue() {
-		if (this.value == null) {
-			return null;
-		}
-
-		this.concreteValue = new ConcreteValue(this.value);
 		return this.concreteValue;
 	}
 
 	public void setConcreteValue(ConcreteValue concreteValue) {
 		this.concreteValue = concreteValue;
 	}
 
-	public void setConcreteValue(String value) {
-		if (value != null) {
-			this.concreteValue = new ConcreteValue(value);
+	public void setConcreteValueFromExternal(org.snomed.otf.owltoolkit.domain.Relationship.ConcreteValue concreteValue) {
+		if (concreteValue != null) {
+			this.concreteValue = ConcreteValue.from(concreteValue);
+			this.value = concreteValue.getRF2Value();
+			this.destinationId = null;
+			this.target = null;
 		}
-
-		this.destinationId = null;
-		this.value = value;
-		this.target = null;
 	}
 
 	public boolean isConcrete() {
-		return this.value != null && this.destinationId == null;
+		final boolean hasValue = this.value != null;
+		final boolean hasConcreteValue = this.concreteValue != null;
+		final boolean hasNoDestinationId = this.destinationId == null || this.destinationId.isEmpty();
+
+		return (hasValue && hasNoDestinationId) || (hasConcreteValue && hasNoDestinationId);
 	}
 
 	@Override
```

Comment:
I think it might be clearer to set destinationId to null in the default constructor and use this.value != null && this.destinationId == null

Reply:
I have made that change @CoderMChu . When I made the change, I was getting a NPE whenever requesting Relationships; so I have added a null check to the expandSourceTypeAndDestination method in RelationshipController.